catkin_tools/jobs/utils.py

```diff
@@ -50,42 +50,67 @@ def get_env_loaders(package, context):
     return sources
 
 
+def merge_envs(job_env, overlay_envs):
+    '''
+    In the merged/linked case of single env, this function amounts to a straight
+    assignment, but a more complex merge is required with isolated result spaces,
+    since a package's build environment may require extending that of multiple
+    other result spaces.
+    '''
+    merge_path_values = {}
+
+    for overlay_env in overlay_envs:
+        for key, values_str in overlay_env.items():
+            if key.endswith('PATH'):
```

Comment:
Oh, why is it necessary in the non-isolated case (just when I thought I was getting a handle on this thing)?

If it is "always on", then yeah I think a whitelist is the best alternative. I'd start with:

- `PATH`
- `CPATH`
- `LD_LIBRARY_PATH`
- `DYLD_LIBRARY_PATH`
- `DYLD_FALLBACK_LIBRARY_PATH`
- `CMAKE_PREFIX_PATH`
- `PKG_CONFIG_PATH`
- `PYTHONPATH`

And any others you can think of.

That being said, I don't think this can be a long term solution. I'm more convinced than ever that this mechanism is just fundamentally broken. I think we'll have to go back to `setup.bash --extend` like mechanics, despite the performance hit. At least I think it should be an option to operate completely with shell scripts, no env parsing/caching/manipulation by `catkin_tools`.